### ISSUE TITLE
Our discussion

### ISSUE THREAD

**Issue by dimakm** [dimakm/GWG-shoppingList#4]
Hello mates,

I closed the first issue and opened this issue, so you don't get confused and you can focus on the new issue.

I just put the starting point and we'll continue to work together and improve our project.
I created a very simple **HTML + CSS + JS** files where we can add an item to a list with the quantity, _for example I put APPLES IN the first input area , and 5 in the second, when I click the button it will add 
APPLES 5  
to the list._
 You can find the files under **<>Code tab** .

To view the files in your browser there's a very simple way, open the index.html , copy the page URL and past it in the page : 
http://htmlpreview.github.io/ 

there's another more complicated way to view the html page in github we'll talk about it later.

If you like to view it in codepen here's the link:
https://codepen.io/Dima-KM/pen/OvjZZe

What I'm thinking about is to add a UNIT drop down menu (a Select Option menu), so we can put  LB, KG, Gallon OR Piece and so we can get 
APPLES 5 LB  

So what do you think about the work so far? and what are your suggestions?

**Comment by dimakm** [comment]
@antofon 
@Eduardo06sp 
@QAMinn2468

**Comment by Eduardo06sp** [comment]
So we aren't using jQuery at all?

I also want to add client-side storage of the list. :}
Will definitely need help with it!

**Comment by dimakm** [comment]
@Eduardo06sp we can use jquery, I'll change the code to make it jQuery.
The client side storage is a good idea but I have no idea how to make it yet.
so if you want to do your research about it it would be great, maybe @antofon can help with that ;)

I'll try to assign tasks so we can all have something to do :

@QAMinn2468 could you please do the CSS styling and make the interface looks good.

@antofon  your turn is coming soon, I just want to change the code to jQuery first and will contact you

**Comment by antofon** [comment]
@Eduardo06sp you want the users to store their list on like a database? Just trying to understand where you're coming from. And okay, @dimakm working on HTML.

**Comment by dimakm** [comment]
@Eduardo06sp we can use jquery, I'll change the code to make it jQuery.
The client side storage is a good idea but I have no idea how to make it yet.
so if you want to do your research about it it would be great, maybe @antofon can help with that ;)

I'll try to assign tasks as soon as I finish the doing the first step in jquery so we all have something to do.

have a nice day!
Dima

**Comment by antofon** [comment]
@dimakm sorry I'm a bit confused? You said my turn is coming soon? So what do you want me working on? The client side storage? I know I had my name down for HTML but looks like you have the basic functionality down.

**Comment by antofon** [comment]
@dimakm Actually it's ok, I understand now. Just re-read the previous messages.

**Comment by antofon** [comment]
I'll wait for you to contact me.

**Comment by antofon** [comment]
@Eduardo06sp Check out this page on Web Storage, didn't even know it was a thing :) https://www.w3schools.com/html/html5_webstorage.asp

**Comment by antofon** [comment]
@Eduardo06sp
[Info from w3schools:](https://www.w3schools.com/html/html5_webstorage.asp)
 
**HTML web storage** provides two objects for storing data on the client:

window.localStorage - stores data with no expiration date
window.sessionStorage - stores data for one session (data is lost when the browser tab is closed)

Do we want to store the data for one session or have it be permanent? If the data stays even after the browser is closed and re-opened, it may be wise to have a 'Clear List' button. Even an **items limit** would be nice (no more than 30 items for example). Maybe have a global count that checks how many items are in the list? Just thinking aloud here.

**Comment by QAMinn2468** [comment]
I am able to do the CSS!  Will pull and take a fresh peek.

Kim (QAMinn2468)

Sent from Mail for Windows 10

**Comment by dimakm** [comment]
@QAMinn2468  thank you! nice colors

**Comment by dimakm** [comment]
### Hello everyone!
Sorry guys if I caused confusion to any of you in the earlier messages :) I was trying to figure out how to start working together as a team, and I was confused too ;) my first project so it made me a little confused.

@antofon sorry for making you confused, I myself was confused a little bit and trying to figure out a way to start the project, I found the best way is by making the basic Javascript/Jquery and then we improve it and work on it together, the HTML is actually simple and we can all work on it as needed for example you need to add a text input field , so go ahead and add it, I need to add a scroll down list, I go ahead and add it, and so on. 

---------------------------------------------------
 **so my friends here are the project's updates:**
 I worked on the code and made it JQuery ,(after suffering for half a day because all the Jquery code I wrote was not working at all then I realized that Jquery itself was having a problem and not working on my laptop and I needed to add this line 

<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>

to the html to make the Jquery work on my computer, you might not need this line though) 

I also added some styles in addition to the styles that @QAMinn2468 already made .the files on github are updated so you can see the changes, or you can go to codepen here's the link:
[Shopping-list codepen](https://codepen.io/Dima-KM/pen/EEwmOO
)I love codepen because I can see the html,css and javascript side by side.

-----------------------------------------------------
 **So far we can only add items to the list.**

-------------------------------------------------------
What we need to to : 
 ### 1- add quantity 
 ### 2-add the unit (KG, LB, oz,gallon ...etc) this can be made as a drop down list and we choose which unit we want.
 ### 3-  the ability to check or strike through the items we already bought.
###  4- the ability to delete items from the list

What else you think we need? could a shopping list have more options ?

 @antofon and @Eduardo06sp  What would you like to work on guys ? 
              Do you think I need to change anything in the Jquery code I made? if yes then what is it?
Eduardo if you are working on client-side storage of the list then you don't have to work on any of the things I listed above.

Have a good night guys

**Comment by QAMinn2468** [comment]
Sounds great!

A shopping app I use is called Out of Milk. It is beyond what we will do,
but may be useful to refer to.

Kim

**Comment by antofon** [comment]
@dimakm Okay no worries. Working on tasks as needed sounds good. I can add the quantity and unit since quite a bit of work has already been completed. Good job with the jQuery, looks so much cleaner :) One suggestion would be to wrap all the code in this statement:


`$(document).ready(function() {
  console.log("DOM content fully loaded.");
// rest of code
});`

It's good practice to check if the DOM is safe to use, check out this [page](https://www.w3schools.com/jquery/event_ready.asp).

@QAMinn2468 great job with the CSS! I have some changes that I was playing with on a forked Codepen as well.

I'll work on my tasks tomorrow since it's pretty late over here. Have a good night everyone!

**Comment by antofon** [comment]
Actually I went ahead did Task 1, I was in the zone haha. Going to commit what I have so far but here's a brief summary:

- Created an alert when users try to input numbers as an item (just some error checking).
- Removed 's' if there is only one item created, even if the user enters it plurally (i.e. 1 eggs -> 1 egg)

I'm also adding comments for increased clarity. Having a lot of fun re-learning JavaScript/jQuery, it's all coming back to me. This project is going great so far.

**Comment by antofon** [comment]
Okay it's super late and I tend to get obsessed when I'm coding. Just submitted a pull request for the quantity option. Hope the code is clear. Now I'm going to sleep.

**Comment by dimakm** [comment]
Good morning everyone,
First day of school after Spring Break, finally my kindergartner is at school today so I can work during the day :) I still have my 20 month old with me at home, but thank God she takes a 2 hours nap so I can work.

@QAMinn2468  I just downloaded the app,  The Out Of Milk creators are having competitors now haha, our shopping list ;) . 
I have a cool shopping app called List Ease, I love its look and style ,the background looks like a real paper .

@antofon great work! I like everything you've done so far. when I do coding I also get excited and want to accomplish something and sometimes I forget about the time. 

I'll work on making the bought items checked . 

Glad to be working with you guys!

**Comment by QAMinn2468** [comment]
I started watching Code College CSS3: 30 projects in 30 days (free).  Very pleased.  The lesson I just started (day 1 “Fancy Buttons” – video #7), discusses toggle switch buttons and how they are not <button> ‘s , but <label> with <input type=”checkbox”>.  A variant of this may be what we are looking for to show the item has been checked off the list.

Kim (QAMinn2468)

Sent from Mail for Windows 10

**Comment by antofon** [comment]
@QAMinn2468 checkbox sounds like the way to go. We could also wrap a strikethrough element <del> or <s> around the shoppingItem variable in the JavaScript. Pretty much toggle the strikethrough 'on' and 'off' depending on the checkbox state ('true or 'false').

**Comment by antofon** [comment]
Woah did not mean to do that haha. I forgot markdown is allowed so it actually did a strikethrough. I mean the tags 'del' or 's'

**Comment by dimakm** [comment]
@QAMinn2468 that's cool!  I made the checkboxes this morning and uploaded it to our github.
@antofon that's what I did today, I added this line:
` <input type='checkbox' name='done' class='itemDone'/>`  to the javascript code that adds `<li>` 

then I made a function **doneBuyItem()**  that toggles the css class **.strike** on and off whenever the check box is clicked. the class  .**strike** is what makes the text has a line through.

 check it out guys and let me know if anything needs to be changed.

**Comment by dimakm** [comment]
@QAMinn2468  Kim I just read your  documentation about project changes 
Thanks to bringing things that need to be worked on to our attention , 

here's Kim's documentation : 
**-Lightened odd list items
-removed outline on active button
-added margin top to button, this balances the vertical spacing

-> attempted to balance horizontal (center the stripes), but to no success.

Noticed:
-input boxes are no longer clearing after submit.  **FIXED by Dima**
-user must choose a number of items
-submitted items with an s, the s is always dropped.**

**Comment by antofon** [comment]
Thanks for the noting the input changes @QAMinn2468. @dimakm suggested that there should be a quantity field, so users can add an amount of an item (i.e. 3 apples). I was trying to drop the 's' on conditions that were of the following:

(x1) breads -> bread 
(x1) cheeses ->cheese
(x1) waters -> water

Also, if a user did not enter a quantity at all, by default the quantity should be 1. 

I will take care of clearing the input boxes after submit, as well as the 's' concern.

**Comment by dimakm** [comment]
@antofon  I'm thinking of something, maybe the user wants a quantity more than 5 , maybe 100 .. and it is hard to write all the numbers up to 100 as  `<option>`
I have a suggestion:  changing the quantity field and make it as an input with a type=number so they can't enter text in it , and we can put limits like min=1 and max=100 for example.. 
I mean something like this `<input type="number" name="quantity" id="quantity" min="1" max="100">`
what do you think about it?

by the way I made the delete button, and I'll upload it now to github, I can't stop working, I love this project and It feels good when I accomplish something in it.
 I also made some css color change when we hover on the buttons.
here's the codepen:
https://codepen.io/Dima-KM/pen/wmrZbV

The basic work is almost done, and we have time to add extra things and more styling to make it more professional. 

Have a good night

**Comment by antofon** [comment]
@dimakm Oh good thinking, I like that. I was wondering what our limit would be, 100 sounds good for now. Later on, we can even add a "Back to top" button if the user doesn't want to scroll all the way back up manually.

The delete button and hover effect are awesome! Subtle change I want to show you. When you hover over, the transition works. But hovering out, the transition ends abruptly. You can do this:

.btn {
    background-color: #00008B;
    color: #FFFAF0;
    font-size: 18px;
    cursor: pointer;
    padding:5px;
    border-radius: 25px;
    width: 10em;
  
  /*fade the color for different crowsers*/
 transition: background-color .7s ease-in-out;
  -webkit-transition: background-color 0.7s ease-in-out;
 -moz-transition: background-color 0.7s ease-in-out;
 -o-transition: background-color 0.7s ease-in-out;
 transition: background-color 0.7s ease-in-out;
}

.btn:hover{ 
   background: #ef4e2e;
}

Initially you have ease-in, but **ease-in-out** gives a slow start and end transition effect. You also want to place the transition effects in the regular .btn class and just place the color you want to change to in .btn:hover

**Comment by antofon** [comment]
Oh you can also do the same thing for the delete class its :hover selector

**Comment by dimakm** [comment]
@antofon thank you for the information, I'll work on the code and do what you said

**Comment by dimakm** [comment]
Hello everyone, hope you all are doing well!

@Eduardo06sp thank you for adding the alert message inside the list, it looks nice that way.

I made some CSS changes, I changed the style a little and I made a background with Adobe illustrator and Photoshop and added it , because I wanted to make our little project more unique so the background with the theme will do that
I used a font from Google Fonts, but I don't know why the font is not working when I open the page from GitHub, but it works on CodePen,  here's the codepen:
https://codepen.io/Dima-KM/pen/wmrZbV

 We still have to add the UNIT and to modify the Quantity @antofon if you haven't worked on them I can do that if you like. I have time during the next few days.

@QAMinn2468 hope you're feeling better. take good care of yourself.

Have a good night everyone.

**Comment by antofon** [comment]
@dimakm Yes if you can add the unit and modify the quantity, that'd be great. Thank you, I've been a bit busy studying for an upcoming coding assessment. I plan to wrap that up by Sunday. But, if anything really needs to get done, I can be flexible.

**Comment by dimakm** [comment]
Hello everyone,
I added the Unit to the grocery list, and worked on modifying the quantity, and now it only removes the 's' from the end of the word if you have 1 item. but it doesn't add 's' if we are buying more than one item, it adds 's' to the unit, example:
(3 gallons) milk
(2 lbs) banana 
kg apple

The reason I didn't add 's' to the item itself, we have some items can't have 's' like milk, we can write it milks, or even rice, can't be rices ... and so on.

I don't want to make it so complicated at this point,we all don't have much time.
 maybe we can improve it in the future.

so what are the final changes need to be done?
I'm planning to put it in the Group Project showcase channel in slack ASAP . so do you suggest any changes before that?

**Comment by antofon** [comment]
@dimakm Looks good to go! Feel free to put it on Slack, like you said, improvements can always be made.

**Comment by dimakm** [comment]
Guys I removed the 's' things for now, there are many cases where having 's' is not correct linguistically, 
Just as @QAMinn2468  mentioned with the pull request she made today, she gave us examples:
3 chicken-whole becomes 3 "s" chicken-whole

so If we want to keep it we need to discuss all the cases in the Javascript code, and it will take alot of work and I know we are all busy. so for now I removed the 's' and later if we want to improve it we'll work on that and discuss all the cases.

The final project is uploaded to github. I'm making a About.html page to include our names in it , and then I'll put everything on Slack. i'll mention your names so you get notifications when someone comments on it on slack

**Comment by dimakm** [comment]
I published the project in slack channel, yaaaay :) 
Thank you guys, I enjoyed working with you.

**Comment by antofon** [comment]
Hooray!!! Thank you for inviting us to work on this project. It's been fun learning about collaboration and applying our skills to actual projects :)

**Comment by glqiao971102** [comment]
Hi guys, I am a Degree in computer science student from Coventry University. I am taking a course which is Open Source Development.  I need to clone your GWG-shopping list project and improve 3 features on it. Please allow me to do it :).

**Comment by Eduardo06sp** [comment]
@glqiao971102 you should go ahead and make any changes then just create a pull request!

@dimakm feel free to give add me as a collaborator if you don't want to manage the repo :}
if not, I recommend you enable Github pages for this repo!
